alpine/alpine-unfixed/alpine.go

```diff
@@ -0,0 +1,202 @@
+		branchPkg := packageData[0]
+		vulnerability := packageData[1]
+		if vulnVersion, exists := branchPkgMap[branchPkg]; exists {
+			if vulns, ok := vulnVersion[version]; ok {
+				vulns = append(vulns, vulnerability)
+				vulnVersion[version] = vulns
+			} else {
+				vulns = []string{vulnerability}
+				vulnVersion[version] = vulns
+			}
+
+		} else {
+			vulns := []string{vulnerability}
+			branchPkgMap[branchPkg] = VulnVersionMap{version: vulns}
+		}
+	}
+	return u.save(branchPkgMap)
+}
+
+func (u Updater) save(packageData map[string]VulnVersionMap) error {
+	for branchPkg, vulnData := range packageData {
+		branchPkgDet := strings.Split(branchPkg, "@@")
+		branch := strings.Split(branchPkgDet[0], "-")
+		release := branch[0]
+		repoName := branch[1]
+		packageName := branchPkgDet[1]
+		dir := filepath.Join(u.vulnListDir, alpineDir, release, repoName)
+		file := fmt.Sprintf("%s.json", packageName)
+		saveJson := SaveJsonFormat{
+			DistroVersion: fmt.Sprintf("v%s", release),
+			RepoName:      repoName,
+			UnfixVersion:  vulnData,
+			PkgName:       packageName,
+		}
+		if err := utils.WriteJSON(u.appFs, dir, file, saveJson); err != nil {
+			return xerrors.Errorf("failed to write %s under %s: %w", file, dir, err)
+		}
+	}
+	return nil
+}
+
+func parseBranchVulnerabilities(branchPath, branch string, baseData map[string]string, baseUrl string) error {
+	b, err := fetchURL(branchPath, "application/ld+json")
+	if err != nil {
+		return err
+	}
+	var releaseInfo ReleaseInfo
+	if err = json.Unmarshal(b, &releaseInfo); err != nil {
+		return err
+	}
+	for _, item := range releaseInfo.Items {
+		_, packageName := path.Split(item.CPEMatch[0].Package)
+		_, vulnerability := path.Split(item.CPEMatch[0].Vulnerability)
+		key := fmt.Sprintf("%s@@%s:%s", branch, packageName, vulnerability)
+		if _, exists := baseData[key]; !exists {
+			err := cveHTMLParser(baseUrl, vulnerability, baseData)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getActiveReleases(url string) ([]string, error) {
+	var activeRelease []string
+	b, err := fetchURL(url, "text/html")
+	if err != nil {
+		return activeRelease, err
+	}
+
+	d, err := goquery.NewDocumentFromReader(bytes.NewReader(b))
+	if err != nil {
+		return activeRelease, err
+	}
+
+	d.Find("h2").Each(func(i int, selection *goquery.Selection) {
+		activeRelease = append(activeRelease, selection.Text())
+	})
+	return activeRelease, nil
+}
+
+func fetchURL(url, acceptHeader string) ([]byte, error) {
+	req := gorequest.New().Get(url)
+	req.Header.Add("Accept", acceptHeader)
+	resp, body, errs := req.Type("text").EndBytes()
+	if len(errs) > 0 {
+		return nil, xerrors.Errorf("HTTP error. url: %s, err: %w", url, errs[0])
+	}
+	if resp.StatusCode != 200 {
+		return nil, xerrors.Errorf("HTTP error. status code: %d, url: %s", resp.StatusCode, url)
+	}
+	return body, nil
+}
+
+func cveHTMLParser(baseUrl, vulnerability string, baseData map[string]string) error {
```

Comment:
Why do we need to parse CVE pages?

Reply:
we have to parse CVE html page. Consider below example.
1. Use https://security.alpinelinux.org/branch/edge-community to all vulnerabilities and packages in edge-community version in json format.
2. Take a CVE(https://security.alpinelinux.org/vuln/CVE-2019-12212) from the above response and check in JSON.

data will be like below screenshot
![image](https://user-images.githubusercontent.com/65420264/128253868-ada8b930-db75-4697-ae8e-447145e1fd58.png)

But if you see same in html response 
![image](https://user-images.githubusercontent.com/65420264/128253951-99b1811a-aff4-4e2b-bfd8-91949afc108d.png)

In JSON you cannot distinguish which state object belongs to which release. But in HTML response, you can clearly see that.

Reply:
This endpoint returns "Potentially vulnerable packages in edge-community". So, you don't need to check the HTML. You already know "freeimage" is potentially vulnerable to "CVE-2019-12212".
https://security.alpinelinux.org/branch/edge-community

Also, you don't need to worry about 3.14-community. CVE-2019-12212 is listed as below.
https://security.alpinelinux.org/branch/3.14-community

Reply:
Yes, but we need to get the package version also.

Reply:
Why? It is not a fixed version.

Reply:
yes it not fixed version. But we want to consider it as affected to version. Means until this version package is vulnerable.